src/cloudstorage/base.py

```diff
@@ -698,6 +701,8 @@ def upload_blob(
         :return: The uploaded blob.
         :rtype: Blob
         """
+        if isinstance(filename, Path):
```

Comment:
Can you also add this check to the other Drivers?

Reply:
It's checked in base.py though, so it should already work for any Driver.

I only added unit tests to test_drivers_amazon.py because the change is not driver-specific. Adding duplicated tests to others will be redundant and slow testing down. Please let me know if you want them added anyway.